apps/accounts/filters.py

```diff
@@ -190,24 +191,41 @@ def filter_execution(queryset, name, value):
         return queryset.filter(**{name: value})
 
 
-class ChangeSecretRecordFilterSet(SecretRecordMixin, UUIDExecutionFilterMixin, BaseFilterSet):
-    execution_id = django_filters.CharFilter(method="filter_execution")
+class DaysExecutionFilterMixin:
     days = drf_filters.NumberFilter(method="filter_days")
+    field: str
 
-    @staticmethod
-    def filter_days(queryset, name, value):
+    def filter_days(self, queryset, name, value):
         value = int(value)
 
         dt = local_zero_hour()
         if value != 1:
             dt = local_now() - timezone.timedelta(days=value)
-        return queryset.filter(date_finished__gte=dt)
+        return queryset.filter(**{f'{self.field}__gte': dt})
+
+
+class ChangeSecretRecordFilterSet(
+    SecretRecordMixin, UUIDExecutionFilterMixin,
+    DaysExecutionFilterMixin, BaseFilterSet
+):
+    execution_id = django_filters.CharFilter(method="filter_execution")
+    days = drf_filters.NumberFilter(method="filter_days")
+
+    field = 'date_finished'
 
     class Meta:
         model = ChangeSecretRecord
         fields = ["id", "status", "asset_id", "execution_id"]
 
 
+class AutomationExecutionFilterSet(DaysExecutionFilterMixin, BaseFilterSet):
+    field = 'date_start'
+
+    class Meta:
+        model = AutomationExecution
+        fields = ["days"]
+
+
 class PushAccountRecordFilterSet(SecretRecordMixin, UUIDExecutionFilterMixin, BaseFilterSet):
     execution_id = django_filters.CharFilter(method="filter_execution")
 
```

Comment:
There may be some issues with the changes you mentioned, especially between version 1 (2021) and version 2 (current). However, I can't see any specific discrepancies that require adjustment at this time. It's important to thoroughly test all functionalities after making any significant changes.

Remember to always double-check your updates! Please let me know if there's something else more concrete or clear you need help on, like testing new versions of the code, documentation updating, etc.